### PR TITLE
Enable release candidates

### LINKED ISSUE #965: release automation should handle RCs
We are planning on making an RC this week, so will need to make sure our release automation can handle this.

### PR DESCRIPTION
**Pull request by errordeveloper**
Fixes #965

### CODE REVIEW

File: do-release-candidate.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh -ex
```
Comment (errordeveloper) [inline do-release-candidate.sh]: I figure I'll make a copy for now, as `do-release.sh` is already complicated and I'd rather not make changes to it yet. We should be able to merge after we get rid of `latest_release` non-sense (see #903).

File: tag-release-candidate.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
```
Comment (martina-if) [inline tag-release-candidate.sh]: Why not reuse the same tag-release script? The only special thing about this one I can see is the release notes file.

Comment (errordeveloper) [reply tag-release-candidate.sh]: There would be a few more conditions. I wanted to keep them separate for now, we can merge later. In fact after we drop `latest_release`, these scripts should become nearly identical.

File: do-release-candidate.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+if [ -z "${CIRCLE_PULL_REQUEST}" ] && [ -n "${CIRCLE_TAG}" ] && [ "${CIRCLE_PROJECT_USERNAME}" = "weaveworks" ] ; then
+  export RELEASE_DESCRIPTION="${CIRCLE_TAG}"
+  RELEASE_NOTES_FILE="docs/release_notes/${CIRCLE_TAG/-rc.*}.md"
+
+  if [[ ! -f "${RELEASE_NOTES_FILE}" ]]; then
+    echo "Release notes ${RELEASE_NOTES_FILE} not found. Exiting..."
+    return
+  fi
+
+  goreleaser release --skip-validate --config=./.goreleaser.yml --release-notes="${RELEASE_NOTES_FILE}"
```
Comment (martina-if) [inline do-release-candidate.sh]: Will this push the RC to brew and other platforms?

File: Makefile [hunk 4]
```diff
@@ -191,18 +191,24 @@ eksctl-image: eksctl-deps-image## Create the eksctl image
 
 ##@ Release
 
+docker_run_release_script = docker run \
+  --env=GITHUB_TOKEN \
+  --env=CIRCLE_TAG \
+  --env=CIRCLE_PROJECT_USERNAME \
+  --volume=$(CURDIR):/src \
+  --workdir=/src \
+    $(EKSCTL_DEPENDENCIES_IMAGE)
```
[the remark this answers is not in the document]
Comment (errordeveloper) [reply Makefile]: Maybe it should, but I wasn't clear to me and I just did what worked.

[the remark this answers is not in the document]
Comment (martina-if) [reply Makefile]: @2opremio

File: do-release-candidate.sh (hunk 3 above)
Comment (errordeveloper) [reply do-release-candidate.sh]: No

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (errordeveloper) [reply Makefile]: Now that I understand the purpose, I'll make use of it.

**Pull request merged**